pepper-apis/src/main/java/org/broadinstitute/ddp/copy/AnswerToAnswerCopier.java

```diff
@@ -0,0 +1,176 @@
+    }
+
+    public void copy(FormResponse sourceInstance, QuestionDto sourceQuestion,
+                     FormResponse targetInstance, QuestionDto targetQuestion) {
+        Answer sourceAnswer = sourceInstance.getAnswer(sourceQuestion.getStableId());
+        if (sourceAnswer != null) {
+            copySourceAnswer(sourceAnswer, targetInstance, targetQuestion);
+        } else {
+            Answer parentAnswer = retrieveParentQuestion(sourceQuestion)
+                    .map(parentDto -> (CompositeAnswer) sourceInstance.getAnswer(parentDto.getStableId()))
+                    .orElse(null);
+            if (parentAnswer != null) {
+                copyFromParentAnswer((CompositeAnswer) parentAnswer, sourceQuestion, targetInstance, targetQuestion);
+            } else {
+                LOG.info("No source answer to copy from activity instance {} and question {}",
+                        sourceInstance.getGuid(), sourceQuestion.getStableId());
+            }
+        }
+    }
+
+    public Optional<CompositeQuestionDto> retrieveParentQuestion(QuestionDto childQuestion) {
+        CompositeQuestionDto parentDto = parentDtosByChildId.get(childQuestion.getId());
+        if (parentDto == null) {
+            parentDto = jdbiCompositeQuestion
+                    .findParentDtoByChildQuestionId(childQuestion.getId())
+                    .orElse(null);
+            if (parentDto != null) {
+                for (var childDto : parentDto.getChildQuestions()) {
+                    parentDtosByChildId.put(childDto.getId(), parentDto);
+                }
+            }
+        }
+        return Optional.ofNullable(parentDto);
+    }
+
+    private void copyFromParentAnswer(CompositeAnswer parentAnswer, QuestionDto sourceQuestion,
+                                      FormResponse targetInstance, QuestionDto targetQuestion) {
+        QuestionDto targetParentQuestion = retrieveParentQuestion(targetQuestion)
+                .orElseThrow(() -> new DDPException(
+                        "Could not find parent composite question for target question " + targetQuestion.getStableId()));
+
+        // Find the target composite answer, creating it if it doesn't exist
+        Answer result = targetInstance.getAnswer(targetParentQuestion.getStableId());
+        if (result == null) {
+            CompositeAnswer ans = new CompositeAnswer(null, targetParentQuestion.getStableId(), null);
+            result = answerDao.createAnswer(operatorId, targetInstance.getId(), targetParentQuestion.getId(), ans);
+            targetInstance.putAnswer(result);
+        }
+        CompositeAnswer targetParentAnswer = (CompositeAnswer) result;
+
+        // Line up the rows between source and target composite answers
+        List<AnswerRow> sourceRows = parentAnswer.getValue();
+        List<AnswerRow> targetRows = targetParentAnswer.getValue();
+        while (targetRows.size() < sourceRows.size()) {
+            targetRows.add(new AnswerRow());
+        }
```

Comment:
Do we need to be concerned about a case where `targetRows.size() > sourceRows.size()` ?

Reply:
I pondered it a bit but not sure how we want to handle it. For now, we're only copying from the size range of the source since that's what's available in the source.